lib/fs/basicfs_walk.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This part copied directly from golang.org/src/path/filepath/path.go (Go
+// 1.6) and lightly modified to be methods on BasicFilesystem.
+
+// In our Walk() all paths given to a WalkFunc() are relative to the
+// filesystem root.
+
+package fs
+
+import (
+	"errors"
+	"path/filepath"
+)
+
+// SkipDir is used as a return value from WalkFuncs to indicate that
+// the directory named in the call is to be skipped. It is not returned
+// as an error by any function.
+var errSkipDir = errors.New("skip this directory")
+var SkipDir = errSkipDir // silences the lint warning...
```

Comment:
Shouldn't this live in filesystem.go?
I suspect all filesystems should return the same error?